### DISCUSSION TITLE
How Can we able to add frameworks using cmakelist file ?

### DISCUSSION THREAD

**Discussion by arungowda09** [axmolengine/axmol#1090] [Q&A] [answered]
recently I have ported game from cocos2dx to axmol. while building for ios I have to add storekit framework.
every time after build it get resets with no framework in Xcode.

I need help to add frameworks using camkefile itself instead of adding everytime after running Xcode build command.

**Comment by rh101** [comment, marked as answer]
Yes, it's possible, and as an example:

    target_link_libraries(${APP_NAME}
        "-framework MediaPlayer"
        "-framework StoreKit"
        )


Also, if you need an entitlements file to enable certain features and such, then let Xcode generate that file, then add that file to your project folder, and in `CMakeLists.txt`, add `XCODE_ATTRIBUTE_CODE_SIGN_ENTITLEMENTS` to the `set_target_properties`, like this:

    set_target_properties(${APP_NAME} PROPERTIES
        RESOURCE "${APP_UI_RES}"
        XCODE_ATTRIBUTE_CODE_SIGN_ENTITLEMENTS "${CMAKE_CURRENT_LIST_DIR}/YourEntitlementFile.entitlements"
        XCODE_ATTRIBUTE_ENABLE_IN_APP_PURCHASE "YES"
        )

As you can see above, you can also add `XCODE_ATTRIBUTE_ENABLE_IN_APP_PURCHASE` if you have in app purchases enabled (I can't remember how I found this, but it does seem to work).

**Reply by arungowda09** [reply]
Thank you @rh101  
also I have 1 more query related to adding CocoaPods for iOS project can you help me with this

**Reply by rh101** [reply]
I don't know anything about CocoaPods, so can't help with that.
